src/controllers/UserController.js

```diff
@@ -147,6 +147,77 @@ class UserController {
     }
   }
 
+  /**
+   *
+   * @description method that updates user's profile settings
+   * @static
+   * @param {object} req HTTP Request object
+   * @param {object} res HTTP Response object
+   * @returns {object} HTTP Response object
+   * @memberof ProfileController
+   */
+  static async updateProfile(req, res) {
+    const userId = req.params.id;
+    try {
+      const user = await User.findOne({ where: { id: userId } });
+      try {
+        const birthDate = HelperMethods.returnDate(req.body.birthDate, user.birthDate);
+        await user.update({
+          firstName: req.body.firstName || user.firstName,
+          lastName: req.body.lastName || user.lastName,
+          password: req.body.password || user.password,
+          gender: req.body.gender || user.gender,
+          birthDate,
+          preferredLanguage:
+            req.body.preferredLanguage || user.preferredLanguage,
+          preferredCurrency:
+            req.body.preferredCurrency || user.preferredCurrency,
+          city: req.body.city || user.city,
+          state: req.body.state || user.state,
+          zip: req.body.zip || user.zip,
+          country: req.body.country || user.country,
+          profileImage: req.body.profileImage || user.profileImage,
+          role: req.body.role || user.role,
+          department: req.body.department || user.department,
+          lineManager: req.body.lineManager || user.lineManager,
+          hasProfile: true,
+        });
```

Comment:
hmm, so lines 165-184 does not look efficient. find a way to only update what the user is sending to the backend. with your current implementation if we add a new column, then we need to come here to add that column when we want to update. please refactor. Reach out if you need help in coming up with the logic for it.